src/app/GitCommands/Settings/AppSettings.cs

```diff
@@ -722,6 +722,41 @@ private static void MigrateSshSettings()
             }
         }
 
+        /// <summary>
+        /// Migrate editor settings if GE was installed in 'Program Files (x86)' before 4.3.
+        /// Only check and update global Git settings in Windows, the only set by GE.
+        /// Guess previous, migrate to current path (i.e. the first usage).
+        /// </summary>
+        private static void MigrateEditorSettings()
+        {
+            // Only run this once
+            bool isMigrated = IsEditorSettingsMigrated;
+            if (isMigrated)
+            {
+                return;
+            }
+
+            EnvironmentConfiguration.SetEnvironmentVariables();
+            ConfigFileSettings configFileGlobalSettings = ConfigFileSettings.CreateGlobal(false);
+
+            string path = configFileGlobalSettings.GetValue("core.editor");
+            if (!path.Contains("Program Files (x86)/GitExtensions", StringComparison.CurrentCultureIgnoreCase))
+            {
+                return;
+            }
+
+#if DEBUG
+            // avoid setting, this may be in the debugger
+            DebugHelpers.Fail($"Update the core.editor path {path} in {configFileGlobalSettings}");
```

Comment:
Please throw a normal exception in order to not risk a hard termination of the app.